app/controllers/miq_ae_class_controller.rb

```diff
@@ -197,6 +198,13 @@ def get_node_info(node)
     x_history_add_item(:id => x_node, :text => @right_cell_text)
   end
 
+  def add_domain_version_message
+    version = @record.version
+    available_version = @record.available_version
+    return if version.nil? || available_version.nil?
+    @version_message = _("Currently running %s version, you have %s version available.") % [version, available_version] if version != available_version
```

Comment:
There should be one method that gets passed the domain and checks the version information and returns either nil or the message string.  The message that contains the domain name can be used in both places.  (No need to maintain two different strings for this.)

Something like this:
```ruby
  def add_all_domains_version_message(domains)
    @version_messages = domains.collect do |domain|
      domain_version_message(domain)
    end.compact
  end

  def domain_version_message(domain)
    version = domain.version
    available_version = domain.available_version
    return if version.nil? || available_version.nil?
    return if version != available_version

    _("Currently running %s version, you have %s version available.") % [version, available_version]
  end
```
Then above where you currently call `add_domain_version_message` you would do this:
`@version_message = domain_version_message(@record) if @record.domain?`